tests/test-edit-flow-notifications.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+class WP_Test_Edit_Flow_Notifications extends WP_UnitTestCase {
+
+	protected static $admin_user_id;
+	protected static $EF_Notifications;
+		
+	public static function wpSetUpBeforeClass( $factory ) {
+		global $edit_flow;
+
+		/**
+		 * `install` is hooked to `admin_init` and `init` is hooked to `init`.
+		 * This means when running these tests, you can encounter a situation
+		 * where the custom post type taxonomy has not been loaded into the database
+		 * since the tests don't trigger `admin_init` and the `install` function is where
+		 * the custom post type taxonomy is loaded into the DB. 
+		 * 
+		 * So make sure we do one cycle of `install` followed by `init` to ensure 
+		 * custom post type taxonomy has been loaded.
+		 */
+		$edit_flow->custom_status->install();
+		$edit_flow->custom_status->init();
+
+		self::$admin_user_id = $factory->user->create( array( 'role' => 'administrator' ) );
+		
+		self::$EF_Notifications = new EF_Notifications();
+		self::$EF_Notifications->install();
+		self::$EF_Notifications->init();
+	}
+
+	public static function wpTearDownAfterClass() {
+		self::delete_user( self::$admin_user_id );
+		self::$EF_Notifications = null;
+	}
+
+	function setUp() {
+		parent::setUp();
+
+		global $pagenow;
+		$pagenow = 'post.php';
+	}
+
+	function tearDown() {
+		parent::tearDown();
```

Comment:
As a best practice, I think it's best to call `parent::tearDown()` after our logic has run (as wp-tests does other resets that we may inadvertently impact).

Reply:
Ah yea, good catch